### ISSUE TITLE
Pixel 8a only really updates location when app is active and phone screen is on

### ISSUE THREAD

**Issue by ChristopheKoeck** [home-assistant/android#4679]
**Home Assistant Android app version(s):**
2024.7.3-full
**Android version(s):**
14 system updated to September 5,2024
**Device model(s):**
Pixel 8a
**Home Assistant version:**
2024.9.3
**Last working Home Assistant release (if known):**
not sure
**Description of problem, include YAML if issue is related to notifications:**
When the app is in the background, or the screen of the phone is turned off location updates happen very random or not, when the app is active and the screen is on i get updates every minute as expected

**Companion App Logs:**

```
There are no logs when the app is running in the background.
I was asked to include screen shots from the Location history in the Android app

```

**Screenshot or video of problem:**
Left the house at 2:55pm, arrived at work at 3pm, app didn't update until 3:08pm when i opened the app
![Driving - App background - Screen off](https://github.com/user-attachments/assets/1155d329-de5d-451f-8a74-b9a1ba03fe9a)

Here i left the app open and the screen on, i get an update every minute, left work at 1:25pm, got home at 1:29 and received all expected updates
![Driving - App open - Screen on](https://github.com/user-attachments/assets/461ab12d-bcf1-41ca-9914-17bab8191f32)

Here i left the app open and screen on while stationary at work. I again get an update every minute as expected.
![Stationary - App open - Screen on](https://github.com/user-attachments/assets/d6406dcc-228c-4a0e-8411-0ac5aef62308)

Here the app is running in the background and screen off. I turned the screen off at 4:22pm and turned the screen back on at 4:52pm
![Stationary - Screen off](https://github.com/user-attachments/assets/0706c6f3-e269-4ccc-a0fc-c86a824c9743)

**Additional information:**

Confirming settings i checked on my Pixel 8a.
- Home assistant app says it's allowed to run in the background.
- App has access to location services at all times
- App battery use is set to Allow background usage and is set to unrestricted

**Comment by dshokouhi** [comment]
relevant forum thread: https://community.home-assistant.io/t/background-location-not-updating-on-google-pixel-8-pro-and-oneplus-10/772887

**Comment by dshokouhi** [comment]
I wonder if #4662 helps this situation more. Will need to wait for the 2024.9.5 beta to be approved by Google to get some valid feedback there. If youd like to provide feedback feel free to join the beta in the play store and wait for the specific version mentioned above to be installed.

**Comment by yashc18** [comment]
can I work on this issue?

**Comment by dshokouhi** [comment]
> can I work on this issue?

Did you see my comment above? May already have a fix in place. Do you have any other ideas for fixes? Feel free to submit.

**Comment by ChristopheKoeck** [comment]
FYI: Somebody on another forum pointed out they had the same problem, but when they turned on High Accuracy Mode it worked for them. So i tried that and when I turned that on for me i stared getting location updates even when the phone screen was off.
To avoid too much battery drain i set the update interval to 60sec. So far that's been working.

**Comment by dshokouhi** [comment]
high accuracy mode actually contains a foreground service which would indeed allow the app to stay active while its going. It also uses GPS directly whereas without high accuracy its using the fused location provider that uses GPS cell and wifi etc...

@ChristopheKoeck can you try the fix in teh beta to see if its working for you?

**Comment by ChristopheKoeck** [comment]
@dshokouhi I just installed the app.
I'm at work now, but i'll report back after my lunch break. That gives me two trips to see if it works or not.

**Comment by dshokouhi** [comment]
interesting that the single accurate location shows up on its own almost an hour later at the end, did you by chance send the `request_location_update` command during that time? 

Also can you confirm the settings under the single accurate location sensor to see if the option include in sensor updates is not checked?

settings > companion app > manage sensors > single accurate location > include in sensor update is off

there is also nothing like data saver enabled on the device too?

**Comment by ChristopheKoeck** [comment]
yes, i have a motion sensor under my house that triggers a request_location_update, that was one of my attempts to get a quicker location update, but it doesn't work well.

i checked and include in sensor update is off.
data saver is not enabled and  i also made sure that the companion app still has full data access even if data saver would turn on

**Comment by dshokouhi** [comment]
thanks for checking, I am definitely not seeing those big gaps in my end. Can you try another test for me and see if you enable the Last Notification sensor and grant its special permission not see if the frequency in updates changes? That permission forces the app to remain awake and may help.

a couple random screenshots from my history where the longest gap was a lil over 10 minutes
![image](https://github.com/user-attachments/assets/4fc051cd-9994-48cc-b918-fb571feea965)
![image](https://github.com/user-attachments/assets/a6fc6295-c6c0-4180-8528-cd42a895665c)
![image](https://github.com/user-attachments/assets/c1abaa1a-47a1-483a-9f93-13949fd469ed)

**Comment by ChristopheKoeck** [comment]
i changed the setting you mentioned, i'll see what that does and report back.
On the bottom pic you posted there are also some bigger gaps
2:36:00 to 2:43:32
2:43:32 to 2:56:12

**Comment by dshokouhi** [comment]
were not concerned with those smaller gaps actually they are to be expected at times. Its teh 30 minutes or longer gaps as we should not see those. We actually had an issue in the past where location updates would stop entirely until we registered for them again (not the same as this one where updates come when the screen turns on, in the previous case they were at a standstill even when the app was open). That check happens about every 15 minutes or so. Should do the evaluation anytime app sensors update.

**Comment by ChristopheKoeck** [comment]
to be clear, if i turn the screen on, but not make the companion app the active app, it still takes a while or i might have to open the app to get the update

**Comment by dshokouhi** [comment]
> to be clear, if i turn the screen on, but not make the companion app the active app, it still takes a while or i might have to open the app to get the update
> 

Are you seeing other sensor updates during these time gaps?

**Comment by ChristopheKoeck** [comment]
you mean any other sensor that's activated in the companion app?
not sure how to check for that.
Could you give an example?

**Comment by dshokouhi** [comment]
Do you see the battery level sensor update in the server during the time gap?

**Comment by ChristopheKoeck** [comment]
I'll check that and let you know

**Comment by ChristopheKoeck** [comment]
Since i enabled the Last Notification sensor i get updates roughly every 5 minutes
I also checked if i would get updates on the battery sensor while the phone screen was off.
I got an update every 15min while the phone was charging, which i guess is normal for that type of sensor.
Here is another Location History log after enabling that sensor
![After Enabling sensor](https://github.com/user-attachments/assets/3f090d3c-247c-48af-ac23-ebc6a453d605)

Because of the short distances that i travel, living on a tiny island, i will probably still have to enable High accuracy mode to get the result i want. Can you turn that mode on/off via an automation?

**Comment by dshokouhi** [comment]
> Can you turn that mode on/off via an automation?

yes there are several on device automations based around zone and connected bluetooth device, and you can also control it via the server

https://companion.home-assistant.io/docs/core/location#high-accuracy-mode

https://companion.home-assistant.io/docs/notifications/notification-commands#high-accuracy-mode

another thing you can try doing is to increase/decrease your zone sizes to see if that helps your situation

My question regarding the battery level sensor was about before you enabled the other sensors, like when you were experiencing an hour or longer gaps in your tracking history. That was when I was curious to see if it was updating because it woudl tell us if the app were truly asleep or if Google was simply not sending us updates. If the app was truly inactive then you wouldnt see those sensors update either.

**Comment by ChristopheKoeck** [comment]
ok, my mistake, i'll turn of the Last Notification sensor again later today when i charge my phone and see what happens

**Comment by ChristopheKoeck** [comment]
It looks like other sensors do update with the last notification sensor turned off.
When i got to work and checked on my laptop the wifi status of my phone was already updated, but it took me turning on the screen to get the location update.

**Comment by ChristopheKoeck** [comment]
i'm playing around with turning on/off high accuracy mode via automations, but the phone doesn't seem to get those commands until i turned on the screen

**Comment by dshokouhi** [comment]
> i'm playing around with turning on/off high accuracy mode via automations, but the phone doesn't seem to get those commands until i turned on the screen

make sure to add the critical format to speed up delivery of those commands

https://companion.home-assistant.io/docs/notifications/critical-notifications#android

**Comment by ChristopheKoeck** [comment]
I've added the critical format, but it's still very hit or miss if my phone receives the message.
I can look at the traces for the automation and it triggered and send the correct command, but it doesn't seem like the phone receives them.

Would it be possible at some point to have an on device automation that turns on High accuracy mode when not connected to wifi?
That's what i'm trying to do with an automation on the server but it's really not working great.

I also noticed when High accuracy mode is turned on, my Pixel 8a has a message on the bottom of my notifications screen that says the Home assistant App is running in the background, when you turn High Accuracy mode off, the message disappears.

**Comment by dshokouhi** [comment]
> I've added the critical format, but it's still very hit or miss if my phone receives the message.
> I can look at the traces for the automation and it triggered and send the correct command, but it doesn't seem like the phone receives them.

need to see the YAML format to confirm



> Would it be possible at some point to have an on device automation that turns on High accuracy mode when not connected to wifi?
> That's what i'm trying to do with an automation on the server but it's really not working great.

no this is why we have teh command so users can take their own logic in their own  hands, if the app supports all these different methods it will never end.



> I also noticed when High accuracy mode is turned on, my Pixel 8a has a message on the bottom of my notifications screen that says the Home assistant App is running in the background, when you turn High Accuracy mode off, the message disappears.

whats the issue here? sounds like its behaving as expected high accuracy mode uses a foreground service so the app can run in the background without interruption.

**Comment by ChristopheKoeck** [comment]
This is the action i'm running.
When the screen is on the message gets received pretty much instantly, when the screen is off, it's very hit or miss, but then as soon as i turn the screen on it receives usually receives the message straight away or completely not. It's very confusing
<img width="223" alt="High Accuracy mode automation" src="https://github.com/user-attachments/assets/7139b314-ddf5-4b42-a1b9-5bdce6f64321">

**Comment by dshokouhi** [comment]
if those commands are not waking the device as they should be then it seems like your device is more aggressive in preventing background apps from running.  I am not sure what we can do at this point but I think the focus should be on why the app is not able to run in the background freely

**Comment by ChristopheKoeck** [comment]
it's been better since i turned off the adaptive battery setting that was mentioned on the Home Assistant forums..
But i can't say it's running completely perfect.
I guess for my needs i will always need to work with High Accuracy Mode, my travel to and from work is so short that the standard 4-5min update interval is too long for me.
I've used a bunch of triggers to turn on/off High Accuracy mode and set the standard time to 60sec. So far that seems to be working pretty well.

**Comment by dshokouhi** [comment]
may or may not have found a potential culprit

https://issuetracker.google.com/issues/370069681

would you say this issue started after the sept pixel update?

**Comment by ChristopheKoeck** [comment]
For me it was happening before September, but the description of the problem looks pretty much the same

**Comment by dshokouhi** [comment]
yea play services can also be at fault here which is what we depend on for it to work

**Comment by didrocks** [comment]
I do confirm (as a Pixel 6a owner) that indeed, position tracking got worse starting september. I think you found a good potential cause! Zones entrance/exit are not detected when the app is in the background.

This was never really precise (I never got background updates event when walking between 1 and 3 minutes interval as described [in the documentation](https://companion.home-assistant.io/docs/core/location/#android-location-sensors) even with clear sky, phone in my hand but screen off, so good potential accuracy and didn’t see rejections of location update after opening the app diagnosis). But for a month or so, this really degraded up to 30-40 minutes between location updates, while moving location and entering/exiting 2 zones.

Of course, I double checked many times the instuctions for troubleshooting location and everything (from data sending to background location, battery savings settings…), so it really sounds like an Android update impacted it. The other sensors are synced even in background though.

Thanks for hanging here and being so responsive @dshokouhi! I know all those location issues due to so many different settings and causes are very complex  to debug and should be quite repetitive.

**Comment by dshokouhi** [comment]
are you guys on the app beta and also updated to android 15 with latest play system updates? the app beta has some location updates and also the OS update could very well impact behavior.

im not seeing very long gaps in time still when i periodically check things.

**Comment by didrocks** [comment]
I don’t use any beta version, here are the info on my setup:
* Home assistant app (2024.9.4-full).
* Android 14 (Last android update was the 5th of September).
* Google play service version: 24.41.34. (Apparently latest on the store).

**Comment by Logicsystem360** [comment]
Hello guys,

I have the same issue but on my Motorola G24 Power with Android 14 (last update from august), Google play version from 1er september 2024.
App is 2024.10.3-full

You can see here :

![image](https://github.com/user-attachments/assets/2114880a-9d9d-4f8a-a759-5c24005224c1)

The only thing did work (but not really) was the persistent notification. But I can't bring back this, I have allowed all the notifications ! Restarted my phone, the app, etc.

Since I have Android 14, the location tracking isn't working very well and all my automation based on the location don't work as they should. 

Do anyone have a solution for this ? It's starting to be annoying since I have the issue since severals months now

**Comment by dshokouhi** [comment]
@Logicsystem360 have you ensured you have all battery optimizations turned off? I think Motorla may have added some extras

https://dontkillmyapp.com/motorola

the above site may point you i the correct direction to determine if you have the app allowed to run freely in the background.

**Comment by Logicsystem360** [comment]
Thanks for your answer. The Home assistant app is already "without restriction" in the parameters. I can't find another parameter which can restrict Home asssitant for saving energy

**Comment by ChristopheKoeck** [comment]
I've updated to Android 15 and i'm using HA 2024.10.4 and HA Android app 2024.10.4.full
In general i'm not seeing any issues anymore but then today i had a random gap of 50min.
All my other updates are withing 2-5min

Sadly enough for me that's still not quick enough since i can get from work to home in under 5min, so i'm still using high accuracy mode while not connected to wifi to make sure my updates happen every minute while driving.

I was thinking of putting a location tag on my quad so when my phone is close to it and connects via bluetooth it would turn on high accuracy mode. Would something like that work?

**Comment by Tux92** [comment]
I update to Android 15 too, but unfortunately no change. I have 10 minutes travel from work to home and the position update only when I arrive and unlook screen

**Comment by JoFranke** [comment]
Same here, I even bought a Pixel 9 Pro, thinking it would fix the issue with a new phone, so Pixel 7 and Pixel 9 both the same issue. I sometimes dont see location updates sent in the log for a long time (maybe 15 minutes), even though the phone is turned on and the app is in forground. Wired.

Too bad I am also not ablte to turn on the 'last notification sensor' as the keeps asking me to grant permissions, even though I did add HA to ther list and granted all permissions for notifications and allowed a few apps in the list.

Hopefully there will be a solution soon. Thanks.

**Comment by dshokouhi** [comment]
> Sadly enough for me that's still not quick enough since i can get from work to home in under 5min, so i'm still using high accuracy mode while not connected to wifi to make sure my updates happen every minute while driving.
> 
> I was thinking of putting a location tag on my quad so when my phone is close to it and connects via bluetooth it would turn on high accuracy mode. Would something like that work?

for your use case i think high accuracy mode may be needed, its a very short amount of time



> I have 10 minutes travel from work to home and the position update only when I arrive and unlook screen

you have also confirmed any and all battery saving settings from your manufacturer? For this short time interval high accuracy mode may also be helpful



> I sometimes dont see location updates sent in the log for a long time (maybe 15 minutes), even though the phone is turned on and the app is in forground. Wired.

that just means google is not sending the app an update, after about 15 minutes the app will re-register for updates which kicks that off again



> Too bad I am also not ablte to turn on the 'last notification sensor' as the companion app keeps asking me to grant permissions, even though I did add HA to the list and granted all permissions for notifications and allowed a few apps in the list.

all you need to do is grant the HA app the permission in the list, as you say you were able to add some apps to the list that means you were able to turn it on. Unless you mean other apps to the notification permission? what device is that? custom ROM? maybe its not translating the permission properly if the app cannot tell its granted. Either way thats a separate issue from here.

............on the subject of zone changes and geofencing i noticed google has some tips here: https://developer.android.com/develop/sensors-and-location/location/geofencing#Troubleshooting

` Having Wi-Fi on can significantly improve the location accuracy, so if Wi-Fi is turned off, your application might never get geofence alerts depending on several settings including the radius of the geofence, the device model, or the Android version. Starting from Android 4.3 (API level 18), we added the capability of “Wi-Fi scan only mode” which allows users to disable Wi-Fi but still get good network location. It’s good practice to prompt the user and provide a shortcut for the user to enable Wi-Fi or Wi-Fi scan only mode if both of them are disabled.`

I personally am in the habit of turning off wifi scanning so i turned it on to try it out. We dont prompt for wifi currently but a good consideration if that is indeed the issue

for those with a short drive this is a good consideration for high accuracy mode given the below time intervals given (note: first time I've seen actual time estimates, usually their docs are ambiguous with terms like few or several etc...)

`Usually the latency is less than 2 minutes, even less when the device has been moving. If Background Location Limits are in effect, the latency is about 2-3 minutes on average. If the device has been stationary for a significant period of time, the latency may increase (up to 6 minutes).`

additionally on the same page it has recommended zone radius:

`For best results, the minimum radius of the geofence should be set between 100 - 150 meters.` 100 is the default for the home zone maybe a slightly larger size is helpful?

**Comment by Tux92** [comment]
Hi guys, I think I found a solution. Now every 5 minutes I got position update.

Go to Settings > Position > Localisation services > enable Find WiFi Networks and Find Bluetooth Devices

This is a translation because I have Android in Italian language, but I upload all the screen so you can find right menu and options:

![IMG_20241117_095607_622~2](https://github.com/user-attachments/assets/72f2f360-40a7-4071-849f-71f9a4846a4a)

![IMG_20241117_095607_625~2](https://github.com/user-attachments/assets/6049b1fc-d527-4270-947a-bc1a26386d80)

![IMG_20241117_095607_346~2](https://github.com/user-attachments/assets/3fd0f868-e67a-4e36-84ba-18ffce465013)

**Comment by dshokouhi** [comment]
> Hi guys, I think I found a solution. Now every 5 minutes I got position update.
> 
> Go to Settings > Position > Localisation services > enable Find WiFi Networks and Find Bluetooth Devices

thats exactly what i found and suggested above :)

**Comment by Tux92** [comment]
> > Hi guys, I think I found a solution. Now every 5 minutes I got position update.
> > Go to Settings > Position > Localisation services > enable Find WiFi Networks and Find Bluetooth Devices
> 
> thats exactly what i found and suggested above :)

Yes, but I found this setting with search function because I came from OnePlus with a android 10 and there these two setting are under WiFi and Bluetooth area. :)

**Comment by didrocks** [comment]
That’s the settings I already have enabled on my Pixel 6a phone since I started to use HA on it, but this didn’t help with the location issue previously described here.

**Comment by JoFranke** [comment]
> > Too bad I am also not ablte to turn on the 'last notification sensor' as the companion app keeps asking me to grant permissions, even though I did add HA to the list and granted all permissions for notifications and allowed a few apps in the list.
> 
> all you need to do is grant the HA app the permission in the list, as you say you were able to add some apps to the list that means you were able to turn it on. Unless you mean other apps to the notification permission? what device is that? custom ROM? maybe its not translating the permission properly if the app cannot tell its granted. Either way thats a separate issue from here.

Its a Pixel 9 Pro with stock android. Yes I think it is a seperate issue, too many issues around.

So I switched to an iPhone 16 now, since I need location updates to run stable, and found the iPhone is doing much better.

**Comment by kloggy** [comment]
At risk of it being a 'me too' post, We have also had these settings enabled 'forever' on our Pixel 7a phones but we both still have the issue described here.

This is incredibly frustrating, so much of my HA relies on location, not just 'home' or 'not home' which could be accomplished using connection to a specific WiFi SSID but if the geolocation stays 'at home' it makes no difference.

**Comment by JoFranke** [comment]
Looks like we are really stuck with that issue rendering home assistant pretty useless, at least for me? How can the developers made aware about this major issue, and how can we get an answer if this will be worked on?

As I mentioned, I switches to an iphone which doesnt have that issue - do other android phones have the samue issue, whats your experiance?

I really cannot handle the iphone and would badly like to go back to android, the message handling on the iphone is such a pia I could not belive it. Thanks.

**Comment by slflowfoon** [comment]
I can confirm that this issue is present on the Samsung S21 SE. Despite having my interval set to update every 300 seconds, the location tracking often shows no updates for 30 minutes and sometimes more. As far as I can tell, this issue only affects Android devices - my partner's iPhone works flawlessly.

Unfortunately, this significantly impacts the reliability of the geolocation feature in Home Assistant at the moment.

**Comment by dshokouhi** [comment]
Looks like other android apps have an issue too. https://github.com/owntracks/android/issues/1851 and https://github.com/owntracks/android/issues/1865

For those who want a workaround please just use high accuracy mode in the meantime it will work and you can adjust the interval.

**Comment by slflowfoon** [comment]
High accuracy mode has the same issue. Interval set to 300 seconds but location not updated for 30 minutes or more.

**Comment by dshokouhi** [comment]
@slflowfoon please double check all samsung battery optimization settings. Battery saving, power saving, data saving etc.. turn it all off for the app please. There may be more settings beyond that too, Samsung likes to add a lot. 

Your answer goes against what all the other apps have discovered, that GPS location is not impacted. Quite possible your issue is related to battery optimization. There can also be log failures if you havent already checked location tracking history. We may need to split your issue as this issue is not about high accuracy mode and would like to keep things on track in here.

**Comment by lucianlazar1983** [comment]
I have the exact same problem o  my Pixel 9 Pro XL. The location update in the background is completely inconsistent and I made sure that no restriction is in place for the companion app. I have also disabled the adaptive battery function but it makes no difference. The only way is to use Hi accuracy but it drains the battery.

**Comment by Jan19901002** [comment]
I have the same Issues, location update is inconsistent, and request_location_update only updates as soon as i open the app.

the phones are 3x s20fe 5g and 1x s24fe, all power saving options are disabled.

it seems there are 3 workarounds

1. high accuracy mode, but sometimes you cant turn that off by command, even with ttl 0 and priority high
2. set persistent connection to always (lot of battery consumption)
3. activate last notification sensor, that works even if the sensor is disabled, but only if the neccessary permission is still given. as soon as is remove the permission, its not working anymore. (that seems to cost also some battery)

i tried the latest beta app, but does not work either

**Comment by damencho** [comment]
Pixel 7 here. I'm using high accuracy mode ot turn on while getting closer to home. And what I have debugged is looking at logs and timings is that location is updated correctly and the desired notification is been triggered as configured. The problem I see is that the notification does not appear on my phone or my android auto. 
The solution to get the notification is to unlock the phone in the area where high accuracy is already on, no need the app to be active or something. 
Seems to me android problem delaying the notification.

**Comment by neoKushan** [comment]
Just to chime in I'm also experiencing this with my son's phone, a Pixel 7a. I have a Pixel 9 Pro without the issue.

On my Son's device all other sensor data is updated regularly - including gecoded location, but the general location is not. It's very frustrating as I have several automations tied to the house being empty or not.

I've ran through all of the suggestions and debugging here, but I'm still at a loss as to the cause.

**Comment by JoFranke** [comment]
Here is what I also noticed - dont know if this was mentioned before:

On my device (Pro 9 fold) location updates work 90% of the time, however, sometimes, it does not update the location for a very long time, sometimes 2 hrs.. e.g. when I enter home, I still see the location from 2hrs before in the comanion app. Opening the comanion app does noit even trigger a location update.

This results in all the automations failing. 

However - when I then open another app which us using location like googe maps, at the same moment instantly the location on the HA companion app is updated as well and all automations suddenly trigger.

So for some reason, the companion app seems not to call the location service sometimes.

**Comment by jeankhawand** [comment]
> Here is what I also noticed - dont know if this was mentioned before:
> 
> On my device (Pro 9 fold) location updates work 90% of the time, however, sometimes, it does not update the location for a very long time, sometimes 2 hrs.. e.g. when I enter home, I still see the location from 2hrs before in the comanion app. Opening the comanion app does noit even trigger a location update.
> 
> This results in all the automations failing. 
> 
> However - when I then open another app which us using location like googe maps, at the same moment instantly the location on the HA companion app is updated as well and all automations suddenly trigger.
> 
> So for some reason, the companion app seems not to call the location service sometimes.

I experience the same issue on my Xiaomi 15 Android (16 BP2A.250605.015) with the 2025-07-01 patch and hassio app version 2025.7.1-full. Before upgrading to Android 15, it worked most of the time. Now, updates can take hours, as if the app is in deep sleep. The trigger seems to be opening the app, after which I see location access indicated by a green dot sometimes it doesn't. Enabling High Accuracy mode doesn't resolve the problem. All battery exceptions are enabled for the companion app, with Wi-Fi and Bluetooth scanning turned on.

**Comment by Morphy99** [comment]
Has anyone tried the Pixel 9 Pro on Android 16 yet to see if it's any better/ worse? I've been putting the update off for a bit as I'm tired of apps not being ready for the latest Android.

**Comment by dshokouhi** [comment]
Pixel 9 Pro XL here, still seeing gaps of 10 minutes or longer in the apps history section. I think for this we need a library update on Googles end to be fixed or a play services update, not really sure where the logic lies on that.

**Comment by lucianlazar1983** [comment]
I have the Pixel 9 pro xl with android 16, same problem, sometimes it does not update for more than 1 hour.--Lucian LazarFrom: Morphy99 ***@***.***>Sent: Tuesday, July 22, 2025 7:29:52 p.m.To: home-assistant/android ***@***.***>Cc: Lucian Lazar ***@***.***>; Comment ***@***.***>Subject: Re: [home-assistant/android] Pixel 8a only really updates location when app is active and phone screen is on (Issue #4679)Morphy99 left a comment (home-assistant/android#4679)Has anyone tried the Pixel 9 Pro on Android 16 yet to see if it's any better/ worse? I've been putting the update off for a bit as I'm tired of apps not being ready for the latest Android.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.
